averbis/core/_rest_client.py

```diff
@@ -811,27 +837,106 @@ def export_text_analysis(
         )
 
     @experimental_api
-    def export_text_analysis_to_cas(self, document_id: str) -> Cas:
+    def export_text_analysis_to_cas(self, document_name: str) -> Cas:
         """
         HIGHLY EXPERIMENTAL API - may soon change or disappear.
 
         Returns an analysis as a UIMA CAS.
         """
-
+        if self.project.client.get_build_info()["specVersion"].startswith("5."):
+            raise OperationNotSupported(
+                "Text analysis export is not supported for platform version 5.x, it is only supported from 6.x onwards."
+            )
+        document_collection = self.project.get_document_collection(self.document_source_name)
+        document_identifier = document_collection.get_document_identifier(document_name)
+        # noinspection PyProtectedMember
         type_system = load_typesystem(
             self.project.client._export_analysis_result_typesystem(
-                self.project.name, self.document_source_name, document_id, self
+                self.project.name, self.document_source_name,
+                document_identifier, self
             )
         )
 
         # noinspection PyProtectedMember
         return load_cas_from_xmi(
             self.project.client._export_analysis_result_to_xmi(
-                self.project.name, self.document_source_name, document_id, self
+                self.project.name, self.document_source_name, document_name, self
             ),
             typesystem=type_system,
         )
 
+    @experimental_api
+    def add_text_analysis_result_to_document(
+            self,
+            source: Union[Cas, Path, IO],
+            document_name: str,
+            mime_type: str = None,
+            typesystem: "TypeSystem" = None
+    ):
+        """
+        HIGHLY EXPERIMENTAL API - may soon change or disappear.
+
+        Add a text analysis result (i.e. annotated content) to a specified document in this process.
+        This process must be a manual or imported process (i.e. without automatic processing via pipeline).
+        There must not already exist a text analysis result associated
+        with this document, use Process.update_text_analysis_for_document() instead.
+
+        The supported file content types (mime_types) are UIMA CAS XMI (application/vnd.uima.cas+xmi),
```

Comment:
Since this list of file formats is valid for different methods, it should be put into a central place and be referenced from the respective methods accepting them. This includes not only new methods but also existing ones.